packages/optimistic-oracle/src/proposer.js

```diff
@@ -286,159 +285,152 @@ class OptimisticOracleProposer {
-        proposer: receipt.events.DisputePrice.returnValues.proposer,
-        disputer: receipt.events.DisputePrice.returnValues.disputer,
-        identifier: this.hexToUtf8(receipt.events.DisputePrice.returnValues.identifier),
-        ancillaryData: receipt.events.DisputePrice.returnValues.ancillaryData,
-        timestamp: receipt.events.DisputePrice.returnValues.timestamp,
-        proposedPrice: receipt.events.DisputePrice.returnValues.proposedPrice
-      };
-      this.logger.info({
-        at: "OptimisticOracleProposer#sendDisputes",
-        message: "Disputed proposal!⛑",
-        priceRequest,
-        proposalPrice,
-        disputePrice,
-        disputeBond,
-        allowedError: this.disputePriceErrorPercent,
-        txnConfig,
-        disputeResult: logResult
-      });
     }
   }
   // Construct settlement transaction and send or return early if an error is encountered.
   async _settleRequest(priceRequest) {
-    // Create the transaction.
+    // Get successful transaction receipt and return value or error.
     const settle = this.optimisticOracleContract.methods.settle(
       priceRequest.requester,
       this.utf8ToHex(priceRequest.identifier),
       priceRequest.timestamp,
       priceRequest.ancillaryData
     );
-
-    // Simple version of inventory management: simulate the transaction and assume that if it fails,
-    // the caller didn't have enough collateral.
-    let payout, gasEstimation;
-    try {
-      [payout, gasEstimation] = await Promise.all([
-        settle.call({ from: this.account }),
-        settle.estimateGas({ from: this.account })
-      ]);
-    } catch (error) {
-      this.logger.error({
-        at: "OptimisticOracleProposer#settleRequests",
-        message: "Cannot settle for unknown reason☹️",
-        priceRequest,
-        error
-      });
-      return;
-    }
     const txnConfig = {
-      from: this.account,
-      gas: Math.min(Math.floor(gasEstimation * this.GAS_LIMIT_BUFFER), this.txnGasLimit),
       gasPrice: this.gasEstimator.getCurrentFastPrice()
     };
-
     this.logger.debug({
       at: "OptimisticOracleProposer#settleRequests",
       message: "Settling proposal or dispute",
       priceRequest,
-      payout,
       txnConfig
     });
+    const transactionResult = await runTransaction(
+      settle,
+      this.account,
+      txnConfig,
+      this.GAS_LIMIT_BUFFER,
+      this.txnGasLimit
+    );
 
-    // Send the transaction or report failure.
-    let receipt;
-    try {
-      receipt = await settle.send(txnConfig);
-    } catch (error) {
-      this.logger.error({
+    // Handle error:
+    if (transactionResult.error) {
+      if (transactionResult.error.type === "call") {
+        this.logger.error({
+          at: "OptimisticOracleProposer#settleRequests",
+          message: "Cannot settle for unknown reason☹️",
+          priceRequest,
+          error: transactionResult.error
+        });
+        return;
+      } else {
```

Comment:
Same suggestion here as above -- only the message differs, so you should be able to get rid of this if and just use a ternary to determine the `message`.